kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -739,11 +739,31 @@ private synchronized void updateConfigurationsInternal(List<ComponentConfigurati
                     try {
                         updateConfigurationInternal(config.getPid(), config.getConfigurationProperties(),
                                 snapshotOnConfirmation);
-                        break;
                     } catch (KuraException e) {
                         logger.warn("Error during updateConfigurations for component " + config.getPid(), e);
                         causes.add(e);
                     }
+                    break;
+                }
+            }
+        }
+
+        // this step creates any not yet existing factory configuration present in configsToUpdate
+        for (ComponentConfiguration config : configsToUpdate) {
+            String factoryPid = null;
+            final Map<String, Object> properties = config.getConfigurationProperties();
+            if (properties != null) {
+                factoryPid = (String) properties.get(ConfigurationAdmin.SERVICE_FACTORYPID);
+            }
+            if (factoryPid != null && !this.allActivatedPids.contains(config.getPid())) {
+                String pid = config.getPid();
+                logger.info("Creating configuration with pid: {} and factory pid: {}", pid, factoryPid);
+                try {
+                    createFactoryConfiguration(factoryPid, pid, properties, false);
+                    configs.add(config);
+                } catch (KuraException e) {
+                    logger.warn("Error creating configuration with pid: " + pid + " and factory pid: {}", factoryPid,
```

Comment:
You can use `Error creating configuration with pid {} and factory pid {}`. No need to mix message formatting with string concatenation